pkg/controller/lifecycle/actuator.go

```diff
@@ -104,6 +105,17 @@ func (a *actuator) Reconcile(ctx context.Context, logger logr.Logger, ex *extens
 		return err
 	}
 
+	lakomProviderConfig := &lakom.LakomConfig{}
+	if _, _, err := a.decoder.Decode(ex.Spec.ProviderConfig.Raw, nil, lakomProviderConfig); err != nil {
+		// Apply default values if provider config has not been provided
+		logger.Error(err, "Could not decode provider config. Using default value `KubeSystemManagedByGardener` for scope")
+		lakomProviderConfig.Scope = ptr.To(lakom.KubeSystemManagedByGardener)
+	}
```

Comment:
Huh, I would have expected that Decode return a non-nil error if the provided data slice is nil.
This pattern is also used in the registry, so I can add a PR to fix this there as well. Nice catch

Reply:
The issue happens just before the decoder.Decode is called when its parameters are evaluated and then trying to read a field from a nil pointer results in the segmentation fault. If you see the same in other places, it will be very nice if you open PRs to fix them.

Reply:
Oh, yes, the actual nil field is the ProviderConfig, my bad

Reply:
Now that I understood correctly, I can see that they've handled the nil ProviderConfig case in the registry as well, so it was an issue just here